File: days/06/solution.py
with open("days/06/input.txt") as f:
    content = f.read()


def part_1():
    for i in range(0, len(content) - 3):
        x = [x for x in content[i:i+4]]
        if len(set(x)) == 4:
            print(i + 4)
            exit()

def part_2():
    for i in range(0, len(content) - 13):
        x = [x for x in content[i:i+14]]
        if len(set(x)) == 14:
            print(i + 14)
            exit()

File: days/06/solution.go
package main

import (
	"fmt"

	"github.com/Tch1b0/polaris/array"
	"github.com/Tch1b0/polaris/input"
)

func getInput() string {
	return input.Read("days/06/input.txt")
}

func unique[T comparable](arr []T) []T {
	n := []T{}
	for _, v := range arr {
		if !array.Some(n, func(x T, _ int) bool { return v == x }) {
			n = append(n, v)
		}
	}

	return n
}

func part1(n int) int {
	c := getInput()

	for i := 0; i < len(c)-(n-1); i++ {
		x := []rune(c[i : i+n])
		if len(unique(x)) == len(x) {
			return i + n
		}
	}
	return -1
}

func part2() int {
	return part1(14)
}

func main() {
	fmt.Printf("PART 1: %d\n", part1(4))

	fmt.Printf("PART 2: %d\n", part2())
}
